### ISSUE TITLE
Improve support for Puppet Types tokens

### ISSUE THREAD

**Issue by rnelson0** [rodjek/puppet-lint#485]
From @hlindberg in #435:

> The list of types is both brittle, and not correct since data types can be namespaced. The Rule for data types is actually the same as for a CLASSREF, this because all references to things like File, User, etc. are also data types, and so are the references to user defined resource types (created with 'define').
> Not sure what puppet-lint does with the CLASSREF that would make it unsuitable for all data types. The problem here is that it is impossible to statically know what a CLASSREF is referring to as that requires loading the actual referenced entity (a resource type in ruby, a data type, type alias, or a user defined resource type).
> The static list in this PR is already behind, and there will be additional types added.

**Comment by rnelson0** [comment]
This is specifically referring to [lexer.rb's KNOWN_TOKENS](https://github.com/rodjek/puppet-lint/blob/master/lib/puppet-lint/lexer.rb#L91) which includes `:TYPE` with a hard-coded list of known types. The list is not exhaustive. Any variable declaration that is preceeded by a word starting with a capital letter may indicate the word is a `:TYPE`.

**Comment by ekohl** [comment]
This is still an issue, as the following test case shows:

```ruby
describe PuppetLint::Lexer do
  before do
    @lexer = PuppetLint::Lexer.new
  end

  context ':TYPE' do
    it 'should match namespaced Data Types' do                                                 
      token = @lexer.tokenise('Stdlib::Absolutepath').first
      expect(token.type).to eq(:TYPE)
      expect(token.value).to eq('Stdlib::Absolutepath')
    end
  end
end
```

```console
$ bundle exec rspec spec/puppet-lint/lexer_spec.rb -e namespaced
Run options: include {:full_description=>/namespaced/}

PuppetLint::Lexer
  :TYPE
    should match namespaced Data Types (FAILED - 1)

Failures:

  1) PuppetLint::Lexer :TYPE should match namespaced Data Types
     Failure/Error: expect(token.type).to eq(:TYPE)
     
       expected: :TYPE
            got: :CLASSREF
     
       (compared using ==)
     
       Diff:
       @@ -1,2 +1,2 @@
       -:TYPE
       +:CLASSREF
       
     # ./spec/puppet-lint/lexer_spec.rb:1386:in `block (3 levels) in <top (required)>'

Finished in 0.00972 seconds (files took 0.12524 seconds to load)
1 example, 1 failure

Failed examples:

rspec ./spec/puppet-lint/lexer_spec.rb:1384 # PuppetLint::Lexer :TYPE should match namespaced Data Types
```
